### PR TITLE
refac(discord): Update metrics and logging

### PR DESCRIPTION
**Pull request by ykamo001**
Update how we send metrics and handle logging during Discord responses. Using more standard metric events to alleviate logic handling in monitoring platforms. Success events end with `"success"`, and failures events with `"failure"`.

We currently need to ignore specific user errors as a result of us not turning off base alert actions. Some alert actions are broken, because the user has deleted the channel/guild/team, or entered an improper id, but because these continue to execute, we continue to log and put them as part of our metrics. When the main problem is fixed, we will not see so many user errors come through.

Resolves: https://github.com/getsentry/team-core-product-foundations/issues/218#issue-2200762819

### CODE REVIEW

File: src/sentry/integrations/discord/client.py [hunk 1]
```diff
@@ -118,13 +122,22 @@ def track_response_data(
...
+            "error": str(error),
+            "extra": extra,
+        }
+
+        if self.integration_type:
+            log_params[str(self.integration_type)] = self.name
+
+        try:
+            logging_context = getattr(self, "logging_context", None)
+            log_params["logging_context"] = logging_context
```
Comment (cathteng) [inline src/sentry/integrations/discord/client.py:138]: when is `logging_context` populated? what does it mean?

File: src/sentry/integrations/discord/client.py [hunk 2]
```diff
@@ -133,44 +146,65 @@ def track_response_data(
...
+            metrics_key,
             sample_rate=1.0,
-            tags={
-                str(self.integration_type): self.name,
-                "status": code,
-                "is_ok": is_ok,
-                "include_in_slo": include_in_slo,
-                "discord_code": discord_error_code,
-            },
+            tags=log_params,
```
Comment (cathteng) [inline src/sentry/integrations/discord/client.py:194]: won't this create a lot of cardinality, and are we concerned about costs? we are potentially adding tags for a lot of different things (`is_user_error`, `discord_error_code_message`, `discord_error_code`, `discord_error_response`, `logging_context`, `str(self.integration_type)`)

File: src/sentry/integrations/discord/client.py (hunk 1 above)
Comment (ykamo001) [reply src/sentry/integrations/discord/client.py:138]: This is carry over from the existing functionality, I think it comes from up the chain

File: src/sentry/integrations/discord/client.py (hunk 2 above)
Comment (cathteng) [reply src/sentry/integrations/discord/client.py:194]: oh if we were already emitting this i guess it's fine

Comment (ykamo001) [reply src/sentry/integrations/discord/client.py:194]: It's a good call out though, do we pay per tag?

Comment (ykamo001) [reply src/sentry/integrations/discord/client.py]: removed the tags as they don't provide any additional context at this moment

**Review by cathteng — APPROVED** [review]
🚀

**Pull request merged**